### PR TITLE
Run Prettify pre-commit hook

### LINKED ISSUE #31: Add automatic code styling
As we're getting more people involved it'd be great to have some ESlint / TSlint / Prettier setup. 

Bonus if run automatically with Github Actions on PR commits 🚀

### PR DESCRIPTION
**Pull request by briangershon**
Run Prettier on the three major sections (contract, web-client, web-server) as a Git pre-commit hook

Closes #31

Includes:

- Add Prettier to the `web-server` project -- and also run Prettier on those files. With minor Prettier tweaks for existing `contract` and `web-client`.
- Install Husky and pre-commit hook to root of project (with a new package.json file) which runs Prettier for all 3 sections.

Note that I run `yarn` install step to ensure all 3 projects have dependencies installed for running Prettier in each project.

### CODE REVIEW

**Comment by briangershon** [comment]
To optimize this, another strategy could be to centralize `prettier` in the root and call that with in the pre-commit hook like this:

```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

npx prettier --config=./contract/.prettierrc.json --ignore-path ./contract/.prettierignore --write contract
npx prettier --config=./web-client/.prettierrc.json --ignore-path ./web-client/.prettierignore --write web-client
npx prettier --config=./web-server/.prettierrc.json --ignore-path ./web-server/.prettierignore --write web-server
```

and then remove Prettier as a dependency in each package.

Though for `web-client` there are additional Vue linting steps in addition to Prettier.

It just might be simpler to keep the 3 packages separate, it'll just mean a slower pre-commit check for new repositories (that don't have all the dependencies installed yet).

I'm sticking with simpler for now.

**Comment by rasmuscnielsen** [comment]
Hey @briangershon 

So I commented on your previous PR with a possible solution to your problem: https://github.com/Developer-DAO/pixel-avatars/pull/59

However in the mean time I actually managed to reduce the "yarn style" time to < 2 sec (on my Mac) by doing some configuration optimization. So maybe styling on pre-commit is actually a viable (or better) option after-all. So perhaps you can ignore my comment on #59 for now.

However: I tried to pull down your branch and I can't seem to get the pre-commit to work. No scripts are run. 

Maybe I'm missing something?

I also tried to `yarn install` the root package.json. With that said I believe it's essential that it's not dependent on each developer installing anything as that would defeat the purpose on the styling being run for everyone whether they want it or not 😄

File: web-server/package.json [hunk 1]
```diff
@@ -19,6 +20,7 @@
         "keccak256": "^1.0.3"
     },
     "devDependencies": {
-        "nodemon": "^2.0.15"
+        "nodemon": "^2.0.15",
+        "prettier": "^2.5.1"
```
Comment (revmischa) [inline web-server/package.json:24]: Maybe prettier should be a top-level dep?

**Comment by briangershon** [comment]
> However: I tried to pull down your branch and I can't seem to get the pre-commit to work. No scripts are run.

- Hey @rasmuscnielsen I had forgotten to add the install piece, this is now added in the root `package.json`. So when you run `yarn` it'll install Husky and also install the hook.

- Though you're right, the Achilles heal is that if a user never runs `yarn` in the root of the project, the hook will never be installed. Plus on a fresh install, I don't like the time it takes to do 3 yarn installs, since it's confusing to the developer why things are taking so long. (in most projects you are forced to `yarn` install since you need all the dependencies installed, but in this case, it's a monorepo and no one is forced to install the root packages)

I'll head back over to #59 and resolve your feedback there. I think the `pre-commit` hook may not work for this particular case.

@revmischa do you think there's another way to get this to work, or should we go with the Github Action approach (#59)?

**Comment by revmischa** [comment]
> > However: I tried to pull down your branch and I can't seem to get the pre-commit to work. No scripts are run.
> 
> Hey @rasmuscnielsen I had forgotten to add the install piece, this is now added in the root `package.json`. So when you run `yarn` it'll install Husky and also install the hook.
> 
> Though you're right, the Achilles heel is that if a user never runs `yarn` in the root of the project, the hook will never be installed. Plus on a fresh install, I don't like the time it takes to do 3 yarn installs, since it's confusing to the developer why things are taking so long.
> 
> (In most projects you are forced to `yarn` install since you need all the dependencies installed, but in this case, it's a monorepo and no one is forced to install the root packages)
> 
> I'll head back over to #59 and resolve your feedback there. I think the `pre-commit` hook may not work for this particular case.
> 
> @revmischa do you think there's another way to get this to work, or should we go with the Github Action approach (#59)?

Husky and prettier should be deps of the root. I don't think there's any reason to have different prettier/eslint configs per-project is there? Just put it all in the root.

**Pull request closed without merging**